### PR TITLE
Upgrade to React 16 lifecycle, remove ControlDate

### PR DESCRIPTION
**Pull request by mbullington**
# Description

**Breaking changes:**
- Bump `react` and `react-dom` `peerDependencies` to `16.14.0`
- Remove `ControlDate` and `validateStartDateBeforeEndDate` as discussed in the tickets below. It's still sparingly used but shouldn't be a function of `mr-ui` because of all the custom styling required. This also lets us remove `moment`.
- Remove `onContentRender` from `MinimumDurationLoader` to simplify the codebase. I determined this wasn't used with the "onContentRender org:mapbox" Github filter.

**Other changes:**
- Remove all `UNSAFE` uses **within `mr-ui` itself**. https://github.com/davidtheclark/react-displace will still throw these warnings every once and a while.
- Remove un-needed `es6-promise` polyfill from the test kitchen as Promise is now supported everywhere.
- Run `npm install` without `ci` 😬 😬 😬 

Closes https://github.com/mapbox/mr-ui/issues/122
Closes https://github.com/mapbox/mr-ui/issues/108
Closes https://github.com/mapbox/mr-ui/issues/91
Closes https://github.com/mapbox/mr-ui/issues/86

Closes https://github.com/mapbox/mr-ui/issues/132
Closes https://github.com/mapbox/mr-ui/issues/125

cc @tristen for review

### CODE REVIEW

File: src/components/minimum-duration-loader/minimum-duration-loader.js [hunk 1]
```diff
@@ -1,87 +1,97 @@
...
+  // If was initially loaded, load immediately.
+  const [wasLoaded] = useState(isLoaded);
 
-export default class MinimumDurationLoader extends React.PureComponent {
-  static propTypes = {
-    children: PropTypes.node,
-    isLoaded: PropTypes.bool,
-    loader: PropTypes.node,
-    minDuration: PropTypes.number,
-    onContentRender: PropTypes.func
```
Comment (tristen) [inline src/components/minimum-duration-loader/minimum-duration-loader.js:14]: Not sure I'm tracking why this was dropped?

Comment (mbullington) [reply src/components/minimum-duration-loader/minimum-duration-loader.js:14]: Oh my bad. I was going to add it back but didn't get around to it!

Comment (tristen) [reply src/components/minimum-duration-loader/minimum-duration-loader.js:14]: 👍 Let's drop it!

**Review by tristen — APPROVED** [review]
Heck of a package-lock but otherwise the changes test well locally and LGTM!

File: src/components/minimum-duration-loader/minimum-duration-loader.js (hunk 1 above)
Comment (mbullington) [reply src/components/minimum-duration-loader/minimum-duration-loader.js:14]: But good call, I hadn't mentioned this in my patch notes.

**Pull request merged**